edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MerticsSerializerTests.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Xunit;
+
+    public class MerticsSerializerTests
+    {
+        readonly Random rand = new Random();
+
+        [Fact]
+        public void SingleMetricSerializes()
+        {
+            Metric testMetric = new Metric(DateTime.UtcNow, this.metrics[0].name, this.rand.NextDouble() * 50, this.metrics[0].tags);
+
+            byte[] data = RawMetric.MetricsToBytes(new Metric[] { testMetric }).ToArray();
```

Comment:
you should use each method "MetricsToBytes" and "BytesToMetrics" to test each other; what if both miss a field, the test will still pass.

Reply:
Add some comments to explain.

Reply:
the data has no meaningful format on its own. Since this isn't a standardized serialization format, there is no spec to adhere to. All we care about is that the raw binary can be converted back to metrics. 

If either serialization or de-serialization drop fields, the test will fail since it will not match the original test value.